src/vs/vscode.proposed.d.ts

```diff
@@ -108,6 +108,17 @@ declare module 'vscode' {
 		getClickCommand?(node: T): string;
 	}
 
+	export interface TextEditor {
+		/**
+		 * Enters snippet mode in the editor with the specified snippet.
+		 *
+		 * @param snippet The snippet to insert in this edit.
+		 * @param options The undo/redo behaviour around this edit. By default, undo stops will be created before and after this edit.
+		 */
+		edit(snippet: SnippetString, options?: { undoStopBefore: boolean; undoStopAfter: boolean; }): void;
+
+	}
```

Comment:
Should editor-api-tests be changed to point to the local `vscode.d.ts` instead of pulling it from `https://raw.githubusercontent.com/Microsoft/vscode/master/src/vs/vscode.d.ts` at build time?